src/setupWorker/start/utils/validateWorkerScope.ts

```diff
@@ -0,0 +1,17 @@
+import { StartOptions } from '../../glossary'
+
+export function validateWorkerScope(
+  registration: ServiceWorkerRegistration,
+  options?: StartOptions,
+): void {
+  if (!options?.quiet && !location.href.startsWith(registration.scope)) {
```

Comment:
According to the spec, the worker's directory and any subdirectories will be controlled by the worker. It looks safe to assume that if the current page's URL doesn't start with the worker's scope then it is outside of it.

Reply:
That's a good point. Well, we can make this validation more granular and compare protocols/hostnames/paths separately. Would that make more sense?

Reply:
that would make sense, but only if we knew which differences are acceptable (the worker would still work despite differences) and which are not (plus maybe inconsistent between browsers)  😅

I think that having a warning even if everything works will be easy to fix in the future, but if there is no warning and the worker doesn't work will be harder for people to report repro steps

Reply:
I think going with the current implementation and adjusting it based on developers' feedback is the right approach here. There may be false positives, which would be considered bugs, and could be fixed and released in patch versions. The amount of issues this warning may solve is way higher than the inconvenience it may cause. 

Basically, the incremental strategy you've suggested above. Love that!